File: game-server/src/main/java/work/fking/pangya/game/model/HoleMode.java
package work.fking.pangya.game.model;

public enum HoleMode {
    FRONT,
    BACK,
    RANDOM,
    SHUFFLE,
    REPEAT,
    SUFFLE_COURSE;

    private static final HoleMode[] VALUES = values();

    public static HoleMode forId(int id) {
        if (id < VALUES.length) {
            return VALUES[id];
        }
        return null;
    }
}

File: game-server/src/main/kotlin/work/fking/pangya/game/model/HoleMode.kt
package work.fking.pangya.game.model

enum class HoleMode {
    FRONT,
    BACK,
    RANDOM,
    SHUFFLE,
    REPEAT,
    SHUFFLE_COURSE;

    companion object {
        private val VALUES = values()
        fun forId(id: Byte): HoleMode? {
            return if (id < VALUES.size) {
                VALUES[id.toInt()]
            } else null
        }
    }
}
